fenix/app/src/main/java/org/mozilla/fenix/debugsettings/store/DebugDrawerNavigationMiddleware.kt

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.debugsettings.store
+
+import androidx.navigation.NavController
+import mozilla.components.lib.state.Middleware
+import mozilla.components.lib.state.MiddlewareContext
+import org.mozilla.fenix.debugsettings.navigation.DEBUG_DRAWER_ROUTE_HOME
+import org.mozilla.fenix.debugsettings.navigation.DEBUG_DRAWER_ROUTE_TAB_TOOLS
+
+/**
+ * Middleware that handles navigation events for the Debug Drawer feature.
+ *
+ * @param navController [NavController] used to execute any navigation actions on the UI.
+ */
+class DebugDrawerNavigationMiddleware(
+    private val navController: NavController,
+) : Middleware<DebugDrawerState, DebugDrawerAction> {
+
+    override fun invoke(
+        context: MiddlewareContext<DebugDrawerState, DebugDrawerAction>,
+        next: (DebugDrawerAction) -> Unit,
+        action: DebugDrawerAction,
+    ) {
+        when (action) {
+            is DebugDrawerAction.NavigateTo.Home -> {
+                navController.navigate(route = DEBUG_DRAWER_ROUTE_HOME)
+            }
+            is DebugDrawerAction.NavigateTo.TabTools -> {
+                navController.navigate(route = DEBUG_DRAWER_ROUTE_TAB_TOOLS)
+            }
+            is DebugDrawerAction.OnBackPressed -> {
+                navController.popBackStack()
+            }
+            else -> {} // no-op
```

Comment:
@boek and I were talking about this, and using `else` blocks in middlewares is something that we should probably avoid moving forward. Instead, we should just include the full list of no-op actions. This allows us to take full advantage of the compile-time exhaustive checking we get from using sealed classes for the action hierarchies. This adds clarity and discoverability for future development because it requires any new actions to be explicitly handled everywhere they might need to be used .

Reply:
That's fair and it's admittedly what I did in the Store, so I'll follow suit here as well 👍